### PR TITLE
Remove draft part 4

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Add mass product attribute update

Issue #171
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/product/src/Application/Controller/Api/Attribute/DeleteProductAttributeAction.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+     * )
+     *
+     * @throws \Exception
+     */
+    public function removeDraftAttribute(
+        AbstractProduct $product,
+        Language $language,
+        AbstractAttribute $attribute
+    ): Response {
+        if ($attribute->getScope()->isGlobal()) {
```
Comment (piotrkreft) [inline module/product/src/Application/Controller/Api/Attribute/DeleteProductAttributeAction.php:91]: It seems like a generic rule. Wouldn't be worth moving to param converter?

File: module/product/src/Application/Controller/Api/Attribute/UpdateProductAttributeAction.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        Language $language,
+        AbstractProduct $product,
+        AbstractAttribute $attribute,
+        Request $request
+    ): Response {
+        $value = $request->request->get('value');
+        $value = $value !== '' ? $value : null;
+
+        $constraint = $this->provider->provide($attribute);
+        if ($attribute->getScope()->isGlobal()) {
```
Comment (piotrkreft) [inline module/product/src/Application/Controller/Api/Attribute/UpdateProductAttributeAction.php:122]: Same as in Ergonode\Product\Application\Controller\Api\Attribute\DeleteProductAttributeAction:: removeDraftAttribute

File: module/product/src/Application/Controller/Api/Attribute/UpdateProductAttributeAction.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+     * @throws \Exception
+     */
+    public function changeDraftAttribute(
+        Language $language,
+        AbstractProduct $product,
+        AbstractAttribute $attribute,
+        Request $request
+    ): Response {
+        $value = $request->request->get('value');
+        $value = $value !== '' ? $value : null;
```
Comment (piotrkreft) [inline module/product/src/Application/Controller/Api/Attribute/UpdateProductAttributeAction.php:119]:
```suggestion
        $value = $value === '' ? null : $value;
```
It's easier to read with no negation so if not a problem please use it like this

File: module/product/src/Application/Model/Product/Attribute/ProductAttributeFormModel.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+     * @Assert\Uuid(strict=true)
+     */
+    public ?string $id = null;
+
+    /**
+     * @Assert\Valid()
+     *
+     * @var AttributeValueFormModel[]
+     */
+    public array $payload;
```
Comment (piotrkreft) [inline module/product/src/Application/Model/Product/Attribute/ProductAttributeFormModel.php:26]:
```suggestion
    public array $payload = [];
```
construct won't be called always as it seems?

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: module/product/src/Application/Controller/Api/Attribute/DeleteProductAttributeAction.php (hunk 1 above)
Comment (rprzedzik) [reply module/product/src/Application/Controller/Api/Attribute/DeleteProductAttributeAction.php:91]: Is on end of refactoring, not shour is this Action stay on code, now is required for compatibility

**Review by piotrkreft — APPROVED** [review]

**Review by wiewiurdp — APPROVED** [review]

**Pull request merged**